src/group_theory/free_abelian_group_finsupp.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import group_theory.free_abelian_group
+import data.finsupp.basic
+
+/-!
+# Isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+
+In this file we construct the canonical isomorphism between `free_abelian_group X` and `X →₀ ℤ`.
+We use this to transport the notion of `support` from `finsupp` to `free_abelian_group`.
+
+## Main declarations
+
+- `free_abelian_group.equiv_finsupp`: group isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+- `free_abelian_group.coeff`: the multiplicity of `x : X` in `a : free_abelian_group X`
+- `free_abelian_group.support`: the finset of `x : X` that occur in `a : free_abelian_group X`
+
+-/
+
+noncomputable theory
+
+open_locale big_operators
+
+namespace free_abelian_group
+variables (X : Type*)
+
+/-- The group homomorphism `free_abelian_group X →+ (X →₀ ℤ)`. -/
+def to_finsupp : free_abelian_group X →+ (X →₀ ℤ) :=
```

Comment:
Why not `abelian_group_to_finsupp` and `finsupp_to_abelian_group`? I can imagine than when searching the library one might not have in mind what you are doing here and a generic `to_finsupp` might be difficult to understand.

Reply:
Good idea. I'll try to implement in in the next few days.

Reply:
The name here is `free_abelian_group.to_finsupp`, which makes sense to me as it enables dot notation

Reply:
There is now dot notation in both directions.